src/main/java/org/nypl/findawayclientlibrary/PlayBookFragment.java

```diff
@@ -359,8 +260,117 @@ public void loadCoverImage() {
 
 
 
+  /* ------------------------------------ DOWNLOAD EVENT HANDLERS ------------------------------------- */
+
+  /**
+   * Change the message on the download button, letting the user know where we are in the downloading progress.
+   */
+  public void redrawDownloadButton(DownloadService.DOWNLOAD_STATUS status) {
+    if (status == null) {
+      return;
+    }
+
+    if (status.equals(DownloadService.DOWNLOAD_STATUS.DOWNLOAD_RUNNING)) {
+      //downloadButton.setText(getString(R.string.pause));
+      return;
+    }
+
+    if (status.equals(DownloadService.DOWNLOAD_STATUS.DOWNLOAD_PAUSED)) {
+      //downloadButton.setText(getString(R.string.resume));
+      return;
+    }
+
+    if (status.equals(DownloadService.DOWNLOAD_STATUS.DOWNLOAD_STOPPED)) {
+      //downloadButton.setText(getString(R.string.start));
+      return;
+    }
+
+    if (status.equals(DownloadService.DOWNLOAD_STATUS.DOWNLOAD_ERROR)) {
+      //downloadButton.setText(getString(R.string.alert));
+      return;
+    }
+  }
+
+
+  /**
+   * Update the progress bar to reflect where we are in the downloading.
+   * While both the primary and secondary progress percentages are being passed in,
+   * ignore the secondary, and only update the primary for now.  Updating the chapters
+   * is happening too much and thrashing the UI.  To use the secondaryProgress as per chapter
+   * download indicators, will want to take the UI update onto a separate thread.
+   *
+   * @param primaryProgress Represents percent of total book download.
+   * @param secondaryProgress Represents percent of currently downloading chapter.
+   */
+  public void redrawDownloadProgress(Integer primaryProgress, Integer secondaryProgress) {
```

Comment:
I would suggesting using `int` instead of `Integer` wherever possible. The latter induces unnecessary [boxing](http://coderevisited.com/boxing-and-unboxing-in-java/).

Reply:
Findaway sdk sends up Integers.  I don't need them as such, my code is fine with ints, but I wanted to be consistent.